js/data/create_bucket.js

```diff
@@ -15,6 +16,10 @@ function createBucket(layer, buffers, collision, indices) {
         return;
     }
 
+    for (var k in layer.layout) {
+        layer.layout[k] = new StyleDeclaration('layout', layer.type, k, layer.layout[k]).value;
```

Comment:
I'm trying this, but when it's calculated in create_bucket for some reason they don't get reevaluated at different zoom levels — i.e. with a style with
``` json
"text-rotate": {
        "stops": [[14,0], [16, 190]]
      }
```
if you load the page at zoom 18, the text is rotated, but if you load it at 14 it isn't — and never flips. Newly added labels *are* rotated at the z-level where they first come in. … :confused: ?